File: LeetCode Practice Problems/202. Happy Number.cpp
class Solution {
public:
    bool isHappy(int n) {
        set<int> ss;
        while(true)
        {
            int s = 0;
            while(n>0)
            {
                int v = n%10;
                s+= v*v;
                n/=10;
            }
            n = s;
            if(n == 1) return true;
            if(ss.find(n) != ss.end()) return false;
            ss.insert(n);
        }
        if(n == 1) return true;
        return false;
    }
};

File: LeetCode/LeetCode Practice Problems/202-Happy Number.java
class Solution {
    public long findSum(long n) {
        long s = 0;
        while(n > 0) {
            s += (n%10)*(n%10);
            n /= 10;
        }
        return s;
    }
    public boolean isHappy(long n) {
        Set<Long> set = new HashSet();
        while(n != 1) {
            long val = findSum(n);
            if(set.contains(val)) {
                return false;
            }
            set.add(val);
            n = val;
        }
        return true;
    }
}
